### PR TITLE
Feature/ss/issue 25 investigate use of hsm library

### PR DESCRIPTION
**Pull request by sastel**
@piebrain this has the HSM changes and assumes that the player is using 2 level navigation.
Notes:
- [ ] I will have to revert the commit that changes the baud rate for code uploads.
- [ ] Usually I clean up the commit history using "git rebase" but the merges that I did make it difficult this time.
- [ ] I have not deleted the old code from main.cpp yet, but only used "#if 0 ... #endif" to remove it with the preprocessor. This was to make merges easier during development and also so that I would have something to refer to.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,14 @@
 CMakeListsPrivate.txt
 .idea/vcs.xml
 
+# esp stack dump debugging
+dump.txt
+EspStackTraceDecoder.jar
+exception.sh
```
Comment (piebrain) [inline .gitignore:7]: Do you care to share these with the rest of us?

File: platformio.ini [hunk 2]
```diff
@@ -26,7 +28,7 @@ platform = espressif8266
 board = esp12e
 framework = arduino
 board_build.f_cpu = 160000000L
-upload_speed = 921600
+upload_speed = 460800
```
Comment (piebrain) [inline platformio.ini:31]: Please revert this change

File: src/fmpJingleIF.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/*
+ * fmpJingleIF.h
+ *
+ * Copyright (c) 2020 Galcom International Inc.
+ *
+ * FM Radio and Player Amplifier Interface for the FMP HSM
```
Comment (piebrain) [inline src/fmpJingleIF.h:6]: This comment seems to be wrong

File: src/fmpHsm.c [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+
+    fmpDebugIF_logEvent(RUNNING_STATE, msg->event);
+
+    switch (msg->event) {
+    case START_EVT:
+        if (me->runningHist) {
+            next = me->runningHist; // transition to previously active substate
+        }
+        else {
+            next = state[FM_RADIO_STATE]; // TODO replace this with call to access NV
```
Comment (piebrain) [inline src/fmpHsm.c:132]: Would be nice if this actually started in the saved state

**Review by piebrain — COMMENTED** [review]
I don't have any problems with the old code still in main.cpp. It's good to have as reference for now.

File: .gitignore (hunk 1 above)
Comment (sastel) [reply .gitignore]: Pushed a fix.

File: src/fmpHsm.c (hunk 4 above)
Comment (sastel) [reply src/fmpHsm.c:132]: I'll have to implement this before merging then.

Comment (sastel) [reply src/fmpHsm.c:132]: Wow I didn't realize that you'd already implemented that in PStore. Thanks! :)
